parsl/monitoring/db_manager.py

```diff
@@ -463,17 +463,28 @@ def start(self,
 
                 """
                 BLOCK_INFO messages
-
                 """
                 block_info_messages = self._get_messages_in_batch(self.pending_block_queue)
+                # Creation of dictionary to store previous block messages to avoid duplicate messages
+                UBM = {}  # type: Dict[Any, Any]
                 if block_info_messages:
                     logger.debug(
                         "Got {} messages from block queue".format(len(block_info_messages)))
-                    # block_info_messages is possibly a nested list of dict (at different polling times)
-                    # Each dict refers to the info of a job/block at one polling time
+                    # block_info_messages is a nested list of dict (at different polling times)
+                    # block_msg is a list of messages (each message is a dict)
+                    # Each dict refers to the info of a job / block at one polling time
                     block_messages_to_insert = []  # type: List[Any]
                     for block_msg in block_info_messages:
-                        block_messages_to_insert.extend(block_msg)
+                        for this_msg in block_msg:
+                            this_id = this_msg['block_id']
```

Comment:
a block ID isn't enough to identify a block in a particular run - block_id is only relative to a particular parsl executor which you can probably find identified in this_msg['executor_label'].

Using just the block ID means that block 0 of one executor and block 0 of another executor will get confused in this code.

You could maybe make this_id a tuple  `(this_msg['executor_label'], this_msg['block_id'])` and index UBM that way.